pyiron_base/generic/object.py

```diff
@@ -33,32 +35,17 @@ def __init__(self, *args, **kwargs):
         self._storage = DataContainer(table_name='storage')
 
     @property
-    def storage(self):
+    def storage(self) -> DataContainer:
         return self._storage
 
-    def to_hdf(self, hdf, group_name=None):
-        """
-        Serialize everything in the `storage` field to HDF5.
-
-        Args:
-            hdf (ProjectHDFio): HDF5 group object.
-            group_name (str, optional): HDF5 subgroup name.
-        """
-        if group_name is not None:
-            hdf = hdf.create_group(group_name)
-        self._storage.to_hdf(hdf=hdf)
-
-    def from_hdf(self, hdf, group_name=None):
-        """
-        Restore the everything in the `storage` field from an HDF5 file.
-
-        Args:
-            hdf (ProjectHDFio): HDF5 group object.
-            group_name (str, optional): HDF5 subgroup name.
-        """
-        if group_name is not None:
-            hdf = hdf.create_group(group_name)
-        self._storage.from_hdf(hdf=hdf)
+    def _to_hdf(self, hdf: ProjectHDFio):
+        self.storage.to_hdf(hdf=hdf)
+
+    def _from_hdf(self, hdf: ProjectHDFio, version: str=None):
+        self.storage.from_hdf(hdf=hdf)
+
+    def _get_hdf_group_name(self):
+        return ""
```

Comment:
```suggestion
```
`HasHDF` checks for `None` explicitly and returning `""` has the same effect as `None`, so I'd just not provide it.

Reply:
Yes, but it's an `@abstractmethod` and Pycharm will whine if I don't define it here...?

Reply:
Oh, yes.  Then I'd say we change that to return `None` by default.